spec.md

```diff
@@ -2339,6 +2350,11 @@ Otherwise `NodeExpandVolume` MUST be called after successful `NodePublishVolume`
 
 If a plugin only supports expansion via the `VolumeExpansion.OFFLINE` capability, then the volume MUST first be taken offline and expanded via `ControllerExpandVolume` (see `ControllerExpandVolume` for more details), and then node-staged or node-published before it can be expanded on the node via `NodeExpandVolume`.
 
+The `staging_target_path` field is not required, for backwards compatibility, but the CO SHOULD supply it
```

Comment:
if this is an optional field drivers will have to continue to contain logic for `if staging_target_path is unset` (since we cant assume all callers will actually use this new field) and thus not actually fixing the problem specified in the description:
>in particular it relieves the need to store some state on the node.

Reply:
This is a good point. 

If we assume that good COs will always set `staging_target_path` for plugins that implement `STAGE_UNSTAGE_VOLUME` then SP logic can be:

* If `staging_target_path` is set (which it should be for all good COs going forward), you know it is the staging path, and you can compare it to `target_path` if it is different you can assume it is the publish path.
* If `staging_target_path` is unset, but you absolutely need it you can fail the operation and say you require this field to operate. This means you will only support resizing for newer clients (given volume expansion is still beta in k8s, and SPs can control what external-resizer sidecar they pair with, this seems like a reasonable approach).

But the question is do we have a case where a good CO wants to legitimately not set `staging_target_path` AND SP will want it?